docs/naming-conventions.md

```diff
@@ -0,0 +1,42 @@
+# Naming Conventions
+
+All source files and variable names should adhere to the guidelines outlined in this document.
+
+## Rust Code
```

Comment:
This seems incomplete. `snake_case` is used for functions, variables, modules, and macros, but `UpperCamelCase` is used for structs, enums, etc. 

Probably should link to [Rust's naming conventions](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html) as well.